### ISSUE TITLE
Form - error text doesn't meet accessibility contrast requirement

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7440]
### Expected Behavior
Form error text should meet contrast requirements

### Actual Behavior
Form error text doesn't meet contrast ratio requirements

### URL, screen shot, or Codepen exhibiting the issue
https://storybook.grommet.io/?path=/story/input-form-field-states--field-states

<img width="1418" alt="Screenshot 2024-11-19 at 4 24 45 PM" src="https://github.com/user-attachments/assets/ef41ab45-eeec-44f1-bed4-7accb2ecb4c7">

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by luketa8** [comment]
Updated [Success Criteria](https://www.figma.com/design/blk8Ej1z6EuxQaFfxckBjm/Untitled?node-id=2-1912&t=elIqkzVDlSHM3006-4)

- Based on asessments for both [Light mode](https://www.figma.com/design/blk8Ej1z6EuxQaFfxckBjm/Untitled?node-id=1-2851&t=elIqkzVDlSHM3006-4) and [Dark mode](https://www.figma.com/design/blk8Ej1z6EuxQaFfxckBjm/Untitled?node-id=3-47&t=elIqkzVDlSHM3006-11) I'm proposing:

### Colors for error messages:

- Light mode: `status-error` (#B30000) 
- Dark mode: `status-critical` (#FF4040)

**Comment by luketa8** [comment]
Adding visual examples

## Light mode
![Screenshot 2025-01-09 at 4 22 23 PM](https://github.com/user-attachments/assets/a74b70a6-dd3e-4bfd-8011-0c240dcc4c0b)

## Dark mode
![Screenshot 2025-01-09 at 4 20 23 PM](https://github.com/user-attachments/assets/dfde3302-8c5c-480a-b51c-9fa4f0ab28ab)

**Comment by luketa8** [comment]
@jcfilben @taysea Ready for review

**Comment by jcfilben** [comment]
Thanks for the proposal! To avoid any issues with backwards compatibility it think ideally rather than changing the mapping of `formField.error.color` from 'status-critical' to 
```
color: {
  light: 'status-error',
  dark: 'status-warning',
}
```
It would be better to maintain the mapping of `formField.error.color` as `status-critical` and change the value of the `status-critical` color. The other place `status-critical` is used is in Notification as the background color (with 10% opacity) and the icon color when the notification status is set to critical.

Ideally `status-error` and `status-critical` should have separate color values. Is there a color that we can use for `status-critical` that does the following:
- Meets 4.5:1 contrast ratio in light and dark modes
- Is different than the `status-error` color
- Can be used for critical notifications while keeping the icon/background background at a 3:1 contrast ratio

**Comment by luketa8** [comment]
I agree it makes sense not to mix up `critical` (which is a status) and `error` (the error message is a part of a form field). So an element with a critical status such a notification would use `status-critical` while an error message, part of a form field, would use `status-error`.

I think the best approach would be to update the value of `status-error` to `#EB0000`. This color meets 4.5:1 contrast ratio in light and dark modes Is different than the `status-critical` color. 


---
### Visual example

![image](https://github.com/user-attachments/assets/e66b0e8e-8e21-4810-97c9-bd38c39e5bf3)

**Comment by jcfilben** [comment]
I agree that from a naming perspective using `status-error` makes more sense and having a separate color from the notification color would be nice.
What I'm worried about is changing the mapping of `formField.error.color` from anything other than `status-critical` may have unintended consequences for people building theme's off of the base theme. For example if someone has a custom theme where `status-error='red'` and `status-critical='pink'`. Currently for FormField errors they be displayed with pink text. If we change `formField.error.color` to map to `status-error`, when the update Grommet versions, suddenly they get red text instead of the pink text. This isn't backwards compatible.

However if we maintain the mapping of `formField.error.color` to `status-critical` and update the color value for status-critical we wont run into this issue. If people are using the base theme as is they will see a different color come through for FormField error text, but since the previous color wasn't meeting WCAG, I think it can be viewed as a bug fix and be safely backwards compatible.

Tagging the @grommet/devs in case there are other opinions here

**Comment by MikeKingdom** [comment]
We need to be careful to keep colors intended as fill colors separate from colors used for text. `status-critical` is something we tend to associate with a status critical icon. It's the fill of an icon. I could see `status-error` being a text color, although at that point I'd argue the symbolic color name isn't semantic enough (e.g. we have `text` and `text-strong` for a reason. Note you also have to distinguish between colors used as backgrounds with foreground text over them.

**Comment by jcfilben** [comment]
> We need to be careful to keep colors intended as fill colors separate from colors used for text.

I agree with this. However, since we are already in a situation where we are using 1 color for both text and as a fill color (`status-critical` is used for form field error text and the critical notification icon & background color) I was worried that trying to change to use a different color name wouldn't be backwards compatible.

**Comment by MikeKingdom** [comment]
I do agree we shouldn't use status-error in light mode and status-warning in dark mode. We need to consistently use a symbolic color name.

**Comment by luketa8** [comment]
Thanks for the review. I'm not sure about the current application of these colors, given that in the storybook, we have examples using `status-critical` [here](https://storybook.grommet.io/?path=/story/input-form-field-states--field-states) and ones using `status-error` [here](https://storybook.grommet.io/?path=/story/input-form-aggregate-validation--aggregate-validation).  

Regarding the backwards compatibility, what do you think about creating a new color named `text-error` or renaming `status-error` to `text-error` and using it as a default error text color? Is there any way to make that backwards compatible?

**Comment by MikeKingdom** [comment]
I don't think we use red as a text color within the HPE theme (partly for that reason.) If you look at those examples in storybook using the HPE theme (change the theme to `hpe` in the top bar of the storybook). In our guidance I think we only show `status-*` colors being used for icons. Are you finding any examples or guidance in the HPE theme that use these colors differently?

We can consider how to improve the base (grommet) theme but I think our primary use case is with the HPE theme.

**Comment by luketa8** [comment]
Yes, the scope of this issue is limited only to Grommet, no change needed in the HPE theme.

**Comment by MikeKingdom** [comment]
> Yes, the scope of this issue is limited only to Grommet, no change needed in the HPE theme.

That's not true depending our chosen solution, especially since the HPE theme doesn't have an issue with red text

**Comment by jcfilben** [comment]
> That's not true depending our chosen solution, especially since the HPE theme doesn't have an issue with red text

In the hpe theme we have `formField.error.color` set to 'text' so if we make changes to `formField.error.color` in the Grommet theme this shouldn't affect the hpe theme since the value will be overridden with 'text'

**Comment by jcfilben** [comment]
> We can consider how to improve the base (grommet) theme but I think our primary use case is with the HPE theme.

Agree primary use case is hpe theme. Currently the form field error text is meeting accessibility requirements in the hpe theme but the grommet theme is not meeting contrast requirements. So this ticket's goal is to find a way to change the base Grommet theme so that it meets accessibility requirements while maintaining backwards compatibility. We shouldn't make a change in this ticket that affects the hpe theme.

**Comment by jcfilben** [comment]
> in the storybook, we have examples using status-critical [here](https://storybook.grommet.io/?path=/story/input-form-field-states--field-states) and ones using status-error [here](https://storybook.grommet.io/?path=/story/input-form-aggregate-validation--aggregate-validation).

The aggregate validation story is built with a custom error message that manually gets passed status-error as the color value.
<img width="398" alt="Image" src="https://github.com/user-attachments/assets/4864e9a5-daee-43dd-a144-67e85bceda6a" />

The other form stories that are using status-critical as the error text color are using the because it is defined in the base Grommet theme as the color to use for `formField.error.color`

**Comment by jcfilben** [comment]
I thought about if there was other approaches we could take that wouldn't affect backwards compatibility. I think the safest path forward at the moment is to still try and maintain the mapping of `formField.error.color` to `status-critical`. Even if we have a new color namespace like `text-error` we could still have the same backwards compatibility issue of someone who changed the status-critical color value in their custom theme expecting that color to be used for FormField errors.

I recognize that keeping this namespace as is isn't ideal. But for the Grommet theme we are restricted by backwards compatibility. I can add this to the v3 wiki to track as a change we would want to make if we do a v3 of Grommet.

I think we should explore the following:

> Is there a color that we can use for status-critical that does the following:
> 
> Meets 4.5:1 contrast ratio in light and dark modes
> Is different than the status-error color
> Can be used for critical notifications while keeping the icon/background background at a 3:1 contrast ratio

**Comment by MikeKingdom** [comment]
Shouldn't we remove references to `status-error` in our storybook? From what I can tell `status-error` was deprecated several years ago.

**Comment by jcfilben** [comment]
> Shouldn't we remove references to status-error in our storybook? From what I can tell status-error was deprecated several years ago.

I didn't realize status-error was deprecated but I would be supportive of removing it from storybook examples. I think the only place we use it is in the Form/AggregateValidation story

**Comment by luketa8** [comment]
I'm proposing `#EB0000` as the new value for `status-critical` 

- Meets 4,5:1 in both modes
- Tested the new hex as icon vs 10% background (#7441) and meets 3:1 in both modes

**Comment by jcfilben** [comment]
I'm good with `#EB0000`

@grommet/devs any concerns?

**Comment by MikeKingdom** [comment]
No concerns here. Looks good

**Comment by jcfilben** [comment]
Marking this as closed, I think we can move forward with this color
